src/AppMenuSearch.cc

```
@@ -0,0 +1,386 @@
+    }
+    m_lastSearchQuery = text;
+
+    // Clear results if search text is too short
+    if (text.length() < 3) {
+        clear(searchMenu);
+        m_lastResults.clear();
+        Q_EMIT repositionRequested();
+        return;
+    }
+
+    if (!m_appMenuModel) {
```

Clang (14.0.6)

[warning] implicit conversion 'Material::AppMenuModel *' -> bool

(readability-implicit-bool-conversion)